### ISSUE TITLE
Create macro to write to *COMPILER-NOISE-STREAM*

### ISSUE THREAD

**Issue by stylewarning** [rigetti/quilc#535]
A lot of time is spent formatting output (especially floating point numbers and instructions) used to trace the compiler. Currently we just set the `*compiler-noise-stream*` to some null output stream when we don't want to see anything. Almost all uses of `*compiler-noise-stream*` use `format`. I propose we do the following:

(1) Rename `*compiler-noise-stream*` to `*compiler-noise*`, to give the subtle hint that it may not be a stream. It will now be a generalized boolean, specifically `(or stream null)`. Document accordingly.

(2) Create a macro that looks something like this:

```
(defmacro format-noise (str &rest args)
  `(progn
      (when *compiler-noise*
        (format *compiler-noise* ,str ,@args))
      (values)))
```

(I suggest using `(values)` just so accidental `nil`s don't float around, not that they're common.)

(3) Change all instances of writing to `*compiler-noise-stream* to use this macro. In cases where something other than `format` is used, do the `(when *compiler-noise*` yourself.

**Comment by stylewarning** [comment]
Oh, and (4) run the test suite before and after the change, and report on the performance difference.

**Comment by appleby** [comment]
Closed by #585
